# ROSALIND Solutions

![ROSALIND Logo](https://rosalind.info/static/img/logo.png)

This repository contains solutions to problems from the [ROSALIND](https://rosalind.info/) website.

- **Total Problems:** 318
- **Solved:** 32

Short description of all the tasks 
(if the task does not open, it means it has not been solved yet, or you forgot to add a link)

## Bioinformatics Armory

| Code | Description                         | Rosalind Link                           | Solution Link                                                 |
|------|-------------------------------------|-----------------------------------------|---------------------------------------------------------------|
| INI  | Introduction to the Bioinformatics Armory | [Link](https://rosalind.info/problems/ini/) | [Link](Bioinformatics_Armory/INI/INI.py) |
| GBK  | GenBank Introduction                | [Link](https://rosalind.info/problems/gbk/) | [Link](Bioinformatics_Armory/GBK/GBK.py) |
| FRMT | Data Formats                        | [Link](https://rosalind.info/problems/frmt/) | [Link](Bioinformatics_Armory/FRMT/FRMT.py) |
| MEME | New Motif Discovery                 | [Link](https://rosalind.info/problems/meme/) | [Link](Bioinformatics_Armory/MEME/MEME.pdf) |
| NEED | Pairwise Global Alignment          | [Link](https://rosalind.info/problems/need/) | [Link](Bioinformatics_Armory/NEED/NEED.py) |
| TFSQ | FASTQ format introduction          | [Link](https://rosalind.info/problems/tfsq/) | [Link](Bioinformatics_Armory/TFSQ/TFSQ.py) |
| PHRE | Read Quality Distribution          | [Link](https://rosalind.info/problems/phre/) | [Link](Bioinformatics_Armory/PHRE/PHRE.py) |
| PTRA | Protein Translation                 | [Link](https://rosalind.info/problems/ptra/) | [Link](Bioinformatics_Armory/PTRA/PTRA.py) |
| FILT | Read Filtration by Quality          | [Link](https://rosalind.info/problems/filt/) | [Link](Bioinformatics_Armory/FILT/FILT.py) |
| RVCO | Complementing a Strand of DNA       | [Link](https://rosalind.info/problems/rvco/) | [Link](Bioinformatics_Armory/RVCO/RVCO.py) |
| SUBO | Suboptimal Local Alignment          | [Link](https://rosalind.info/problems/subo/) | [Link](Bioinformatics_Armory/SUBO/SUBO.py) |
| BPHR | Base Quality Distribution           | [Link](https://rosalind.info/problems/bphr/) | [Link](Bioinformatics_Armory/BPHR/BPHR.py) |
| CLUS | Global Multiple Alignment           | [Link](https://rosalind.info/problems/clus/) | [Link](Bioinformatics_Armory/CLUS/CLUS.py) |
| ORFR | Finding Genes with ORFs             | [Link](https://rosalind.info/problems/orfr/) | [Link](Bioinformatics_Armory/ORFR/ORFR.py) |
| BFIL | Base Filtration by Quality          | [Link](https://rosalind.info/problems/bfil/) | [Link](Bioinformatics_Armory/BFIL/BFIL.py) |


## Bioinformatics Stronghold 

| Code | Description                         | Rosalind Link | Solution Link |
|------|-------------------------------------|---------------|---------------|
| DNA  | Counting DNA Nucleotides            | [Link](#) | [Link](#) |
| RNA  | Transcribing DNA into RNA           | [Link](#) | [Link](#) |
| REVC | Complementing a Strand of DNA       | [Link](#) | [Link](#) |
| FIB  | Rabbits and Recurrence Relations   | [Link](#) | [Link](#) |
| GC   | Computing GC Content                | [Link](#) | [Link](#) |
| HAMM | Counting Point Mutations            | [Link](#) | [Link](#) |
| IPRB | Mendel's First Law                  | [Link](#) | [Link](#) |
| PROT | Translating RNA into Protein        | [Link](#) | [Link](#) |
| SUBS | Finding a Motif in DNA              | [Link](#) | [Link](#) |
| CONS | Consensus and Profile               | [Link](#) | [Link](#) |
| FIBD | Mortal Fibonacci Rabbits           | [Link](#) | [Link](#) |
| PERM | Enumerating Gene Orders             | [Link](#) | [Link](#) |
| PRTM | Calculating Protein Mass            | [Link](#) | [Link](#) |
| REVP | Locating Restriction Sites          | [Link](#) | [Link](#) |
| GRPH | Overlap Graphs                      | [Link](#) | [Link](#) |
| IEV  | Calculating Expected Offspring      | [Link](#) | [Link](#) |
| LCSM | Finding a Shared Motif              | [Link](#) | [Link](#) |
| LIA  | Independent Alleles                 | [Link](#) | [Link](#) |
| MPRT | Finding a Protein Motif             | [Link](#) | [Link](#) |
| MRNA | Inferring mRNA from Protein         | [Link](#) | [Link](#) |
| ORF  | Open Reading Frames                | [Link](#) | [Link](#) |
| SPLC | RNA Splicing                        | [Link](#) | [Link](#) |
| LEXF | Enumerating k-mers Lexicographically| [Link](#) | [Link](#) |
| PROB | Introduction to Random Strings       | [Link](#) | [Link](#) |
| SIGN | Enumerating Oriented Gene Orderings | [Link](#) | [Link](#) |
| LGIS | Longest Increasing Subsequence      | [Link](#) | [Link](#) |
| LONG | Genome Assembly as Shortest Superstring | [Link](#) | [Link](#) |
| PMCH | Perfect Matchings and RNA Secondary Structures | [Link](#) | [Link](#) |
| PPER | Partial Permutations                | [Link](#) | [Link](#) |
| SSEQ | Finding a Spliced Motif             | [Link](#) | [Link](#) |
| TRAN | Transitions and Transversions       | [Link](#) | [Link](#) |
| TREE | Completing a Tree                   | [Link](#) | [Link](#) |
| CAT  | Catalan Numbers and RNA Secondary Structures | [Link](#) | [Link](#) |
| CORR | Error Correction in Reads          | [Link](#) | [Link](#) |
| INOD | Counting Phylogenetic Ancestors    | [Link](#) | [Link](#) |
| KMER | k-Mer Composition                  | [Link](#) | [Link](#) |
| KMP  | Speeding Up Motif Finding          | [Link](#) | [Link](#) |
| LCSQ | Finding a Shared Spliced Motif      | [Link](#) | [Link](#) |
| LEXV | Ordering Strings of Varying Length Lexicographically | [Link](#) | [Link](#) |
| MMCH | Maximum Matchings and RNA Secondary Structures | [Link](#) | [Link](#) |
| PDST | Creating a Distance Matrix          | [Link](#) | [Link](#) |
| REAR | Reversal Distance                   | [Link](#) | [Link](#) |
| RSTR | Matching Random Motifs             | [Link](#) | [Link](#) |
| SSET | Counting Subsets                    | [Link](#) | [Link](#) |
| ASPC | Introduction to Alternative Splicing | [Link](#) | [Link](#) |
| EDIT | Edit Distance                       | [Link](#) | [Link](#) |
| EVAL | Expected Number of Restriction Sites | [Link](#) | [Link](#) |
| MOTZ | Motzkin Numbers and RNA Secondary Structures | [Link](#) | [Link](#) |
| NWCK | Distances in Trees                  | [Link](#) | [Link](#) |
| SCSP | Interleaving Two Motifs             | [Link](#) | [Link](#) |
| SETO | Introduction to Set Operations      | [Link](#) | [Link](#) |
| SORT | Sorting by Reversals                | [Link](#) | [Link](#) |
| SPEC | Inferring Protein from Spectrum     | [Link](#) | [Link](#) |
| TRIE | Introduction to Pattern Matching    | [Link](#) | [Link](#) |
| CONV | Comparing Spectra with the Spectral Convolution | [Link](#) | [Link](#) |
| CTBL | Creating a Character Table         | [Link](#) | [Link](#) |
| DBRU | Constructing a De Bruijn Graph      | [Link](#) | [Link](#) |
| EDTA | Edit Distance Alignment            | [Link](#) | [Link](#) |
| FULL | Inferring Peptide from Full Spectrum | [Link](#) | [Link](#) |
| INDC | Independent Segregation of Chromosomes | [Link](#) | [Link](#) |
| ITWV | Finding Disjoint Motifs in a Gene   | [Link](#) | [Link](#) |
| LREP | Finding the Longest Multiple Repeat | [Link](#) | [Link](#) |
| NKEW | Newick Format with Edge Weights     | [Link](#) | [Link](#) |
| RNAS | Wobble Bonding and RNA Secondary Structures | [Link](#) | [Link](#) |
| AFRQ | Counting Disease Carriers           | [Link](#) | [Link](#) |
| CSTR | Creating a Character Table from Genetic Strings | [Link](#) | [Link](#) |
| CTEA | Counting Optimal Alignments         | [Link](#) | [Link](#) |
| CUNR | Counting Unrooted Binary Trees      | [Link](#) | [Link](#) |
| GLOB | Global Alignment with Scoring Matrix | [Link](#) | [Link](#) |
| PCOV | Genome Assembly with Perfect Coverage | [Link](#) | [Link](#) |
| PRSM | Matching a Spectrum to a Protein    | [Link](#) | [Link](#) |
| QRT  | Quartets                            | [Link](#) | [Link](#) |
| SGRA | Using the Spectrum Graph to Infer Peptides | [Link](#) | [Link](#) |
| SUFF | Encoding Suffix Trees               | [Link](#) | [Link](#) |
| CHBP | Character-Based Phylogeny            | [Link](#) | [Link](#) |
| CNTQ | Counting Quartets                   | [Link](#) | [Link](#) |
| EUBT | Enumerating Unrooted Binary Trees   | [Link](#) | [Link](#) |
| GASM | Genome Assembly Using Reads         | [Link](#) | [Link](#) |
| GCON | Global Alignment with Constant Gap Penalty | [Link](#) | [Link](#) |
| LING | Linguistic Complexity of a Genome   | [Link](#) | [Link](#) |
| LOCA | Local Alignment with Scoring Matrix | [Link](#) | [Link](#) |
| MEND | Inferring Genotype from a Pedigree  | [Link](#) | [Link](#) |
| MGAP | Maximizing the Gap Symbols of an Optimal Alignment | [Link](#) | [Link](#) |
| MREP | Identifying Maximal Repeats         | [Link](#) | [Link](#) |
| MULT | Multiple Alignment                 | [Link](#) | [Link](#) |
| PDPL | Creating a Restriction Map          | [Link](#) | [Link](#) |
| ROOT | Counting Rooted Binary Trees        | [Link](#) | [Link](#) |
| SEXL | Sex-Linked Inheritance              | [Link](#) | [Link](#) |
| SPTD | Phylogeny Comparison with Split Distance | [Link](#) | [Link](#) |
| WFMD | The Wright-Fisher Model of Genetic Drift | [Link](#) | [Link](#) |
| ALPH | Alignment-Based Phylogeny           | [Link](#) | [Link](#) |
| ASMQ | Assessing Assembly Quality with N50 and N75 | [Link](#) | [Link](#) |
| CSET | Fixing an Inconsistent Character Set | [Link](#) | [Link](#) |
| EBIN | Wright-Fisher's Expected Behavior   | [Link](#) | [Link](#) |
| FOUN | The Founder Effect and Genetic Drift | [Link](#) | [Link](#) |
| GAFF | Global Alignment with Scoring Matrix and Affine Gap Penalty | [Link](#) | [Link](#) |
| GREP | Genome Assembly with Perfect Coverage and Repeats | [Link](#) | [Link](#) |
| OAP  | Overlap Alignment                   | [Link](#) | [Link](#) |
| QRTD | Quartet Distance                    | [Link](#) | [Link](#) |
| SIMS | Finding a Motif with Modifications  | [Link](#) | [Link](#) |
| SMGB | Semiglobal Alignment                | [Link](#) | [Link](#) |
| KSIM | Finding All Similar Motifs          | [Link](#) | [Link](#) |
| LAFF | Local Alignment with Affine Gap Penalty | [Link](#) | [Link](#) |
| OSYM | Isolating Symbols in Alignments     | [Link](#) | [Link](#) |
| RSUB | Identifying Reversing Substitutions | [Link](#) | [Link](#) |




