# Rosalind Solutions

![ROSALIND Logo](https://rosalind.info/static/img/logo.png)

This repository contains solutions to problems from the [ROSALIND](https://rosalind.info/) website. In the solutions, both Python and C++ code were used for algorithms, as well as third-party bioinformatics programs. 

- **Total Problems:** 318
- **Solved:** 70

| Track                       | Status |
|-----------------------------|--------|
| Python Village              | ☑      |
| Bioinformatics Stronghold   | ☐      |
| Bioinformatics Armory       | ☑      |
| Bioinformatics Textbook Track | ☐      |
| Algorithmic Heights         | ☐      |


Short description of all the tasks 
(if the task does not open, it means it has not been solved yet, or you forgot to add a link)


## Bioinformatics Armory

| Code | Description                         | Rosalind Link                           | Solution Link                                                 |
|------|-------------------------------------|-----------------------------------------|---------------------------------------------------------------|
| INI  | Introduction to the Bioinformatics Armory | [Link](https://rosalind.info/problems/ini/) | [Link](Bioinformatics_Armory/INI/INI.py) |
| GBK  | GenBank Introduction                | [Link](https://rosalind.info/problems/gbk/) | [Link](Bioinformatics_Armory/GBK/GBK.py) |
| FRMT | Data Formats                        | [Link](https://rosalind.info/problems/frmt/) | [Link](Bioinformatics_Armory/FRMT/FRMT.py) |
| MEME | New Motif Discovery                 | [Link](https://rosalind.info/problems/meme/) | [Link](Bioinformatics_Armory/MEME/MEME.pdf) |
| NEED | Pairwise Global Alignment          | [Link](https://rosalind.info/problems/need/) | [Link](Bioinformatics_Armory/NEED/NEED.md) |
| TFSQ | FASTQ format introduction          | [Link](https://rosalind.info/problems/tfsq/) | [Link](Bioinformatics_Armory/TFSQ/TFSQ.py) |
| PHRE | Read Quality Distribution          | [Link](https://rosalind.info/problems/phre/) | [Link](Bioinformatics_Armory/PHRE/PHRE.py) |
| PTRA | Protein Translation                 | [Link](https://rosalind.info/problems/ptra/) | [Link](Bioinformatics_Armory/PTRA/PTRA.py) |
| FILT | Read Filtration by Quality          | [Link](https://rosalind.info/problems/filt/) | [Link](Bioinformatics_Armory/FILT/FILT.py) |
| RVCO | Complementing a Strand of DNA       | [Link](https://rosalind.info/problems/rvco/) | [Link](Bioinformatics_Armory/RVCO/RVCO.py) |
| SUBO | Suboptimal Local Alignment          | [Link](https://rosalind.info/problems/subo/) | [Link](Bioinformatics_Armory/SUBO/SUBO.py) |
| BPHR | Base Quality Distribution           | [Link](https://rosalind.info/problems/bphr/) | [Link](Bioinformatics_Armory/BPHR/BPHR.py) |
| CLUS | Global Multiple Alignment           | [Link](https://rosalind.info/problems/clus/) | [Link](Bioinformatics_Armory/CLUS/CLUS.py) |
| ORFR | Finding Genes with ORFs             | [Link](https://rosalind.info/problems/orfr/) | [Link](Bioinformatics_Armory/ORFR/ORFR.py) |
| BFIL | Base Filtration by Quality          | [Link](https://rosalind.info/problems/bfil/) | [Link](Bioinformatics_Armory/BFIL/BFIL.py) |


## Algorithmic Heights

| Code | Description                         | Rosalind Link                                 | Solution Link                                 |
|------|-------------------------------------|-----------------------------------------------|-----------------------------------------------|
| FIBO | Fibonacci Numbers                  | [Link](https://rosalind.info/problems/fibo/) | [Link](Algorithms_Heights/FIBO/FIBO.py) |
| BINS | Binary Search                      | [Link](https://rosalind.info/problems/bins/) | [Link](Algorithms_Heights/BINS/BINS.py) |
| DEG  | Degree Array                       | [Link](https://rosalind.info/problems/deg/)  | [Link](Algorithms_Heights/DEG/DEG.py)  |
| INS  | Insertion Sort                     | [Link](https://rosalind.info/problems/ins/)  | [Link](Algorithms_Heights/INS/INS.py)  |
| DDEG | Double-Degree Array                | [Link](https://rosalind.info/problems/ddeg/) | [Link](Algorithms_Heights/DDEG/DDEG.py) |
| MAJ  | Majority Element                   | [Link](https://rosalind.info/problems/maj/)  | [Link](Algorithms_Heights/MAJ/MAJ.py)  |
| MER  | Merge Two Sorted Arrays            | [Link](https://rosalind.info/problems/mer/)  | [Link](Algorithms_Heights/MER/MER.py)  |
| 2SUM | 2SUM                              | [Link](https://rosalind.info/problems/2sum/) | [Link](Algorithms_Heights/2SUM/2SUM.py) |
| BFS  | Breadth-First Search               | [Link](https://rosalind.info/problems/bfs/)  | [Link](Algorithms_Heights/BFS/BFS.py)  |
| CC   | Connected Components              | [Link](https://rosalind.info/problems/cc/)   | [Link](Algorithms_Heights/CC/CC.py)   |
| MS   | Merge Sort                         | [Link](https://rosalind.info/problems/ms/)   | [Link](Algorithms_Heights/MS/MS.py)   |
| 3SUM | 3SUM                              | [Link](https://rosalind.info/problems/3sum/) | [Link](Algorithms_Heights/3SUM/3SUM.py) |
| DIJ  | Dijkstra's Algorithm               | [Link](https://rosalind.info/problems/dij/)  | [Link](Algorithms_Heights/DIJ/DIJ.py)  |
| INV  | Counting Inversions                | [Link](https://rosalind.info/problems/inv/)  | [Link](Algorithms_Heights/INV/INV.py)  |
| SQ   | Square in a Graph                  | [Link](https://rosalind.info/problems/sq/)   | [Link](Algorithms_Heights/SQ/SQ.py)   |
| HEA  | Building a Heap                    | [Link](https://rosalind.info/problems/hea/)  | [Link](Algorithms_Heights/HEA/HEA.py)  |
| PAR  | 2-Way Partition                    | [Link](https://rosalind.info/problems/par/)  | [Link](Algorithms_Heights/PAR/PAR.py)  |
| BIP  | Testing Bipartiteness              | [Link](https://rosalind.info/problems/bip/)  | [Link](Algorithms_Heights/BIP/BIP.py)  |
| DAG  | Testing Acyclicity                 | [Link](https://rosalind.info/problems/dag/)  | [Link](Algorithms_Heights/DAG/DAG.py)  |
| BF   | Bellman-Ford Algorithm             | [Link](https://rosalind.info/problems/bf/)   | [Link](Algorithms_Heights/BF/BF.py)   |
| CTE  | Shortest Cycle Through a Given Edge | [Link](https://rosalind.info/problems/cte/)  | [Link](Algorithms_Heights/CTE/CTE.py)  |
| HS   | Heap Sort                          | [Link](https://rosalind.info/problems/hs/)   | [Link](Algorithms_Heights/HS/HS.py)   |
| PAR3 | 3-Way Partition                    | [Link](https://rosalind.info/problems/par3/) | [Link](Algorithms_Heights/PAR3/PAR3.py) |
| MED  | Median                             | [Link](https://rosalind.info/problems/med/)  | [Link](Algorithms_Heights/MED/MED.py)  |
| PS   | Partial Sort                        | [Link](https://rosalind.info/problems/ps/)   | [Link](Algorithms_Heights/PS/PS.py)   |
| TS   | Topological Sorting                 | [Link](https://rosalind.info/problems/ts/)   | [Link](Algorithms_Heights/TS/TS.py)   |
| HDAG | Hamiltonian Path in DAG            | [Link](https://rosalind.info/problems/hdag/) | [Link](Algorithms_Heights/HDAG/HDAG.py) |
| NWC  | Negative Weight Cycle               | [Link](https://rosalind.info/problems/nwc/)  | [Link](Algorithms_Heights/NWC/NWC.py)  |
| QS   | Quick Sort                          | [Link](https://rosalind.info/problems/qs/)   | [Link](Algorithms_Heights/QS/QS.py)   |
| SCC  | Strongly Connected Components      | [Link](https://rosalind.info/problems/scc/)  | [Link](Algorithms_Heights/SCC/SCC.py)  |
| 2SAT | 2-Satisfiability                   | [Link](https://rosalind.info/problems/2sat/) | [Link](Algorithms_Heights/2SAT/2SAT.py) |
| GS   | General Sink                        | [Link](https://rosalind.info/problems/gs/)   | [Link](Algorithms_Heights/GS/GS.py)   |
| SC   | Semi-Connected Graph               | [Link](https://rosalind.info/problems/sc/)   | [Link](Algorithms_Heights/SC/SC.py)   |
| SDAG | Shortest Paths in DAG               | [Link](https://rosalind.info/problems/sdag/) | [Link]



## Bioinformatics Stronghold 

| Code | Description                         | Rosalind Link                                 | Solution Link                                 |
|------|-------------------------------------|-----------------------------------------------|-----------------------------------------------|
| DNA  | Counting DNA Nucleotides            | [Link](https://rosalind.info/problems/dna/)   | [Link](Bioinformatics_Stronghold/DNA/DNA.py)   |
| RNA  | Transcribing DNA into RNA           | [Link](https://rosalind.info/problems/rna/)   | [Link](Bioinformatics_Stronghold/RNA/RNA.py)   |
| REVC | Complementing a Strand of DNA       | [Link](https://rosalind.info/problems/revc/)  | [Link](Bioinformatics_Stronghold/REVC/REVC.py) |
| FIB  | Rabbits and Recurrence Relations   | [Link](https://rosalind.info/problems/fib/)   | [Link](Bioinformatics_Stronghold/FIB/FIB.py)   |
| GC   | Computing GC Content                | [Link](https://rosalind.info/problems/gc/)    | [Link](Bioinformatics_Stronghold/GC/GC.py)     |
| HAMM | Counting Point Mutations            | [Link](https://rosalind.info/problems/hamm/)  | [Link](Bioinformatics_Stronghold/HAMM/HAMM.py) |
| IPRB | Mendel's First Law                  | [Link](https://rosalind.info/problems/iprb/)  | [Link](Bioinformatics_Stronghold/IPRB/IPRB.py) |
| PROT | Translating RNA into Protein        | [Link](https://rosalind.info/problems/prot/)   | [Link](Bioinformatics_Stronghold/PROT/PROT.py) |
| SUBS | Finding a Motif in DNA              | [Link](https://rosalind.info/problems/subs/)  | [Link](Bioinformatics_Stronghold/SUBS/SUBS.py) |
| CONS | Consensus and Profile               | [Link](https://rosalind.info/problems/cons/)  | [Link](Bioinformatics_Stronghold/CONS/CONS.py) |
| FIBD | Mortal Fibonacci Rabbits           | [Link](https://rosalind.info/problems/fibd/)  | [Link](Bioinformatics_Stronghold/FIBD/FIBD.py) |
| PERM | Enumerating Gene Orders             | [Link](https://rosalind.info/problems/perm/)  | [Link](Bioinformatics_Stronghold/PERM/PERM.py) |
| PRTM | Calculating Protein Mass            | [Link](https://rosalind.info/problems/prtm/)  | [Link](Bioinformatics_Stronghold/PRTM/PRTM.py) |
| REVP | Locating Restriction Sites          | [Link](https://rosalind.info/problems/revp/)  | [Link](Bioinformatics_Stronghold/REVP/REVP.py) |
| GRPH | Overlap Graphs                      | [Link](https://rosalind.info/problems/grph/)  | [Link](Bioinformatics_Stronghold/GRPH/GRPH.py) |
| IEV  | Calculating Expected Offspring      | [Link](https://rosalind.info/problems/iev/)   | [Link](Bioinformatics_Stronghold/IEV/IEV.py)   |
| LCSM | Finding a Shared Motif              | [Link](https://rosalind.info/problems/lcsm/)  | [Link](Bioinformatics_Stronghold/LCSM/LCSM.py) |
| LIA  | Independent Alleles                 | [Link](https://rosalind.info/problems/lia/)   | [Link](Bioinformatics_Stronghold/LIA/LIA.py)   |
| MPRT | Finding a Protein Motif             | [Link](https://rosalind.info/problems/mprt/)  | [Link](Bioinformatics_Stronghold/MPRT/MPRT.py) |
| MRNA | Inferring mRNA from Protein         | [Link](https://rosalind.info/problems/mrna/)  | [Link](Bioinformatics_Stronghold/MRNA/MRNA.py) |
| ORF  | Open Reading Frames                | [Link](https://rosalind.info/problems/orf/)   | [Link](Bioinformatics_Stronghold/ORF/ORF.py)   |
| SPLC | RNA Splicing                        | [Link](https://rosalind.info/problems/splc/)  | [Link](Bioinformatics_Stronghold/SPLC/SPLC.py) |
| LEXF | Enumerating k-mers Lexicographically| [Link](https://rosalind.info/problems/lexf/)  | [Link](Bioinformatics_Stronghold/LEXF/LEXF.py) |
| PROB | Introduction to Random Strings       | [Link](https://rosalind.info/problems/prob/)  | [Link](Bioinformatics_Stronghold/PROB/PROB.py) |
| SIGN | Enumerating Oriented Gene Orderings | [Link](https://rosalind.info/problems/sign/)  | [Link](Bioinformatics_Stronghold/SIGN/SIGN.py) |
| LGIS | Longest Increasing Subsequence      | [Link](https://rosalind.info/problems/lgis/)  | [Link](Bioinformatics_Stronghold/LGIS/LGIS.py) |
| LONG | Genome Assembly as Shortest Superstring | [Link](https://rosalind.info/problems/long/)  | [Link](Bioinformatics_Stronghold/LONG/LONG.py) |
| PMCH | Perfect Matchings and RNA Secondary Structures | [Link](https://rosalind.info/problems/pmch/)  | [Link](Bioinformatics_Stronghold/PMCH/PMCH.py) |
| PPER | Partial Permutations                | [Link](https://rosalind.info/problems/pper/)  | [Link](Bioinformatics_Stronghold/PPER/PPER.py) |
| SSEQ | Finding a Spliced Motif             | [Link](https://rosalind.info/problems/sseq/)  | [Link](Bioinformatics_Stronghold/SSEQ/SSEQ.py) |
| TRAN | Transitions and Transversions       | [Link](https://rosalind.info/problems/tran/)  | [Link](Bioinformatics_Stronghold/TRAN/TRAN.py) |
| TREE | Completing a Tree                   | [Link](https://rosalind.info/problems/tree/)  | [Link](Bioinformatics_Stronghold/TREE/TREE.py) |
| CAT  | Catalan Numbers and RNA Secondary Structures | [Link](https://rosalind.info/problems/cat/)   | [Link](Bioinformatics_Stronghold/CAT/CAT.py)   |
| CORR | Error Correction in Reads          | [Link](https://rosalind.info/problems/corr/)  | [Link](Bioinformatics_Stronghold/CORR/CORR.py) |
| INOD | Counting Phylogenetic Ancestors    | [Link](https://rosalind.info/problems/inod/)  | [Link](Bioinformatics_Stronghold/INOD/INOD.py) |
| KMER | k-Mer Composition                  | [Link](https://rosalind.info/problems/kmer/)  | [Link](Bioinformatics_Stronghold/KMER/KMER.py) |
| KMP  | Speeding Up Motif Finding          | [Link](https://rosalind.info/problems/kmp/)   | [Link](Bioinformatics_Stronghold/KMP/KMP.py)   |
| LCSQ | Finding a Shared Spliced Motif      | [Link](https://rosalind.info/problems/lcsq/)  | [Link](Bioinformatics_Stronghold/LCSQ/LCSQ.py) |
| LEXV | Ordering Strings of Varying Length Lexicographically | [Link](https://rosalind.info/problems/lexv/)  | [Link](Bioinformatics_Stronghold/LEXV/LEXV.py) |
| MMCH | Maximum Matchings and RNA Secondary Structures | [Link](https://rosalind.info/problems/mmch/)  | [Link](Bioinformatics_Stronghold/MMCH/MMCH.py) |
| PDST | Creating a Distance Matrix          | [Link](https://rosalind.info/problems/pdst/)  | [Link](Bioinformatics_Stronghold/PDST/PDST.py) |
| REAR | Reversal Distance                   | [Link](https://rosalind.info/problems/rear/)  | [Link](Bioinformatics_Stronghold/REAR/REAR.py) |
| RSTR | Matching Random Motifs             | [Link](https://rosalind.info/problems/rstr/)  | [Link](Bioinformatics_Stronghold/RSTR/RSTR.py) |
| SSET | Counting Subsets                    | [Link](https://rosalind.info/problems/sset/)  | [Link](Bioinformatics_Stronghold/SSET/SSET.py) |
| ASPC | Introduction to Alternative Splicing | [Link](https://rosalind.info/problems/aspc/)  | [Link](Bioinformatics_Stronghold/ASPC/ASPC.py) |
| EDIT | Edit Distance                       | [Link](https://rosalind.info/problems/edit/)  | [Link](Bioinformatics_Stronghold/EDIT/EDIT.py) |
| EVAL | Expected Number of Restriction Sites | [Link](https://rosalind.info/problems/eval/)  | [Link](Bioinformatics_Stronghold/EVAL/EVAL.py) |
| MOTZ | Motzkin Numbers and RNA Secondary Structures | [Link](https://rosalind.info/problems/motz/)  | [Link](Bioinformatics_Stronghold/MOTZ/MOTZ.py) |
| NWCK | Distances in Trees                  | [Link](https://rosalind.info/problems/nwck/)  | [Link](Bioinformatics_Stronghold/NWCK/NWCK.py) |
| SCSP | Interleaving Two Motifs             | [Link](https://rosalind.info/problems/scsp/)  | [Link](Bioinformatics_Stronghold/SCSP/SCSP.py) |
| SETO | Introduction to Set Operations      | [Link](https://rosalind.info/problems/seto/)  | [Link](Bioinformatics_Stronghold/SETO/SETO.py) |
| SORT | Sorting by Reversals                | [Link](https://rosalind.info/problems/sort/)  | [Link](Bioinformatics_Stronghold/SORT/SORT.py) |
| SPEC | Inferring Protein from Spectrum     | [Link](https://rosalind.info/problems/spec/)  | [Link](Bioinformatics_Stronghold/SPEC/SPEC.py) |
| TRIE | Introduction to Pattern Matching    | [Link](https://rosalind.info/problems/trie/)  | [Link](Bioinformatics_Stronghold/TRIE/TRIE.py) |
| CONV | Comparing Spectra with the Spectral Convolution | [Link](https://rosalind.info/problems/conv/)  | [Link](Bioinformatics_Stronghold/CONV/CONV.py) |
| CTBL | Creating a Character Table         | [Link](https://rosalind.info/problems/ctbl/)  | [Link](Bioinformatics_Stronghold/CTBL/CTBL.py) |
| DBRU | Constructing a De Bruijn Graph      | [Link](https://rosalind.info/problems/dbru/)  | [Link](Bioinformatics_Stronghold/DBRU/DBRU.py) |
| EDTA | Edit Distance Alignment            | [Link](https://rosalind.info/problems/edta/)  | [Link](Bioinformatics_Stronghold/EDTA/EDTA.py) |
| FULL | Inferring Peptide from Full Spectrum | [Link](https://rosalind.info/problems/full/)  | [Link](Bioinformatics_Stronghold/FULL/FULL.py) |
| INDC | Independent Segregation of Chromosomes | [Link](https://rosalind.info/problems/indc/)  | [Link](Bioinformatics_Stronghold/INDC/INDC.py) |
| ITWV | Finding Disjoint Motifs in a Gene   | [Link](https://rosalind.info/problems/itwv/)  | [Link](Bioinformatics_Stronghold/ITWV/ITWV.py) |
| LREP | Finding the Longest Multiple Repeat | [Link](https://rosalind.info/problems/lrep/)  | [Link](Bioinformatics_Stronghold/LREP/LREP.py) |
| NKEW | Newick Format with Edge Weights     | [Link](https://rosalind.info/problems/nkew/)  | [Link](Bioinformatics_Stronghold/NKEW/NKEW.py) |
| RNAS | Wobble Bonding and RNA Secondary Structures | [Link](https://rosalind.info/problems/rnas/)  | [Link](Bioinformatics_Stronghold/RNAS/RNAS.py) |
| AFRQ | Counting Disease Carriers           | [Link](https://rosalind.info/problems/afrq/)  | [Link](Bioinformatics_Stronghold/AFRQ/AFRQ.py) |
| CSTR | Creating a Character Table from Genetic Strings | [Link](https://rosalind.info/problems/cstr/)  | [Link](Bioinformatics_Stronghold/CSTR/CSTR.py) |
| CTEA | Counting Optimal Alignments         | [Link](https://rosalind.info/problems/ctea/)  | [Link](Bioinformatics_Stronghold/CTEA/CTEA.py) |
| CUNR | Counting Unrooted Binary Trees      | [Link](https://rosalind.info/problems/cunr/)  | [Link](Bioinformatics_Stronghold/CUNR/CUNR.py) |
| GLOB | Global Alignment with Scoring Matrix | [Link](https://rosalind.info/problems/glob/)  | [Link](Bioinformatics_Stronghold/GLOB/GLOB.py) |
| PCOV | Genome Assembly with Perfect Coverage | [Link](https://rosalind.info/problems/pcov/)  | [Link](Bioinformatics_Stronghold/PCOV/PCOV.py) |
| PRSM | Matching a Spectrum to a Protein    | [Link](https://rosalind.info/problems/prsm/)  | [Link](Bioinformatics_Stronghold/PRSM/PRSM.py) |
| QRT  | Quartets                            | [Link](https://rosalind.info/problems/qrt/)   | [Link](Bioinformatics_Stronghold/QRT/QRT.py)   |
| SGRA | Using the Spectrum Graph to Infer Peptides | [Link](https://rosalind.info/problems/sgra/)  | [Link](Bioinformatics_Stronghold/SGRA/SGRA.py) |
| SUFF | Encoding Suffix Trees               | [Link](https://rosalind.info/problems/suff/)  | [Link](Bioinformatics_Stronghold/SUFF/SUFF.py) |
| CHBP | Character-Based Phylogeny            | [Link](https://rosalind.info/problems/chbp/)  | [Link](Bioinformatics_Stronghold/CHBP/CHBP.py) |
| CNTQ | Counting Quartets                   | [Link](https://rosalind.info/problems/cntq/)  | [Link](Bioinformatics_Stronghold/CNTQ/CNTQ.py) |
| EUBT | Enumerating Unrooted Binary Trees   | [Link](https://rosalind.info/problems/eubt/)  | [Link](Bioinformatics_Stronghold/EUBT/EUBT.py) |
| GASM | Genome Assembly Using Reads         | [Link](https://rosalind.info/problems/gasm/)  | [Link](Bioinformatics_Stronghold/GASM/GASM.py) |
| GCON | Global Alignment with Constant Gap Penalty | [Link](https://rosalind.info/problems/gcon/)  | [Link](Bioinformatics_Stronghold/GCON/GCON.py) |
| LING | Linguistic Complexity of a Genome   | [Link](https://rosalind.info/problems/ling/)  | [Link](Bioinformatics_Stronghold/LING/LING.py) |
| LOCA | Local Alignment with Scoring Matrix | [Link](https://rosalind.info/problems/loca/)  | [Link](Bioinformatics_Stronghold/LOCA/LOCA.py) |
| MEND | Inferring Genotype from a Pedigree  | [Link](https://rosalind.info/problems/mend/)  | [Link](Bioinformatics_Stronghold/MEND/MEND.py) |
| MGAP | Maximizing the Gap Symbols of an Optimal Alignment | [Link](https://rosalind.info/problems/mgap/)  | [Link](Bioinformatics_Stronghold/MGAP/MGAP.py) |
| MREP | Identifying Maximal Repeats         | [Link](https://rosalind.info/problems/mrep/)  | [Link](Bioinformatics_Stronghold/MREP/MREP.py) |
| MULT | Multiple Alignment                 | [Link](https://rosalind.info/problems/mult/)  | [Link](Bioinformatics_Stronghold/MULT/MULT.py) |
| PDPL | Creating a Restriction Map          | [Link](https://rosalind.info/problems/pdpl/)  | [Link](Bioinformatics_Stronghold/PDPL/PDPL.py) |
| ROOT | Counting Rooted Binary Trees        | [Link](https://rosalind.info/problems/root/)  | [Link](Bioinformatics_Stronghold/ROOT/ROOT.py) |
| SEXL | Sex-Linked Inheritance              | [Link](https://rosalind.info/problems/sexl/)  | [Link](Bioinformatics_Stronghold/SEXL/SEXL.py) |
| SPTD | Phylogeny Comparison with Split Distance | [Link](https://rosalind.info/problems/sptd/)  | [Link](Bioinformatics_Stronghold/SPTD/SPTD.py) |
| WFMD | The Wright-Fisher Model of Genetic Drift | [Link](https://rosalind.info/problems/wfmd/)  | [Link](Bioinformatics_Stronghold/WFMD/WFMD.py) |
| ALPH | Alignment-Based Phylogeny           | [Link](https://rosalind.info/problems/alph/)  | [Link](Bioinformatics_Stronghold/ALPH/ALPH.py) |
| ASMQ | Assessing Assembly Quality with N50 and N75 | [Link](https://rosalind.info/problems/asmq/)  | [Link](Bioinformatics_Stronghold/ASMQ/ASMQ.py) |
| CSET | Fixing an Inconsistent Character Set | [Link](https://rosalind.info/problems/cset/)  | [Link](Bioinformatics_Stronghold/CSET/CSET.py) |
| EBIN | Wright-Fisher's Expected Behavior   | [Link](https://rosalind.info/problems/ebin/)  | [Link](Bioinformatics_Stronghold/EBIN/EBIN.py) |
| FOUN | The Founder Effect and Genetic Drift | [Link](https://rosalind.info/problems/foun/)  | [Link](Bioinformatics_Stronghold/FOUN/FOUN.py) |
| GAFF | Global Alignment with Scoring Matrix and Affine Gap Penalty | [Link](https://rosalind.info/problems/gaff/)  | [Link](Bioinformatics_Stronghold/GAFF/GAFF.py) |
| GREP | Genome Assembly with Perfect Coverage and Repeats | [Link](https://rosalind.info/problems/grep/)  | [Link](Bioinformatics_Stronghold/GREP/GREP.py) |
| OAP  | Overlap Alignment                   | [Link](https://rosalind.info/problems/oap/)   | [Link](Bioinformatics_Stronghold/OAP/OAP.py)   |
| QRTD | Quartet Distance                    | [Link](https://rosalind.info/problems/qrtd/)  | [Link](Bioinformatics_Stronghold/QRTD/QRTD.py) |
| SIMS | Finding a Motif with Modifications  | [Link](https://rosalind.info/problems/sims/)  | [Link](Bioinformatics_Stronghold/SIMS/SIMS.py) |
| SMGB | Semiglobal Alignment                | [Link](https://rosalind.info/problems/smgb/)  | [Link](Bioinformatics_Stronghold/SMGB/SMGB.py) |
| KSIM | Finding All Similar Motifs          | [Link](https://rosalind.info/problems/ksim/)  | [Link](Bioinformatics_Stronghold/KSIM/KSIM.py) |
| LAFF | Local Alignment with Affine Gap Penalty | [Link](https://rosalind.info/problems/laff/)  | [Link](Bioinformatics_Stronghold/LAFF/LAFF.py) |
| OSYM | Isolating Symbols in Alignments     | [Link](https://rosalind.info/problems/osym/)  | [Link](Bioinformatics_Stronghold/OSYM/OSYM.py) |
| RSUB | Identifying Reversing Substitutions | [Link](https://rosalind.info/problems/rsub/)  | [Link](Bioinformatics_Stronghold/RSUB/RSUB.py) |



